### ISSUE TITLE
Splitter#k_fold_split interface

### ISSUE THREAD

**Issue by lilleswing** [deepchem/deepchem#677]
What this does currently is returns the cross validation set for each of the folds -- but it doesn't return the indices it used or the training sets.

This is useless, if you were to use it you would want BOTH the training and validation datasets.

Can we change this to return a list of (train_dataset, test_dataset) tuples?

```python
  def k_fold_split(self, dataset, k, directories=None, **kwargs):
    """Does K-fold split of dataset."""
    log("Computing K-fold split", self.verbose)
    if directories is None:
      directories = [tempfile.mkdtemp() for _ in range(k)]
    else:
      assert len(directories) == k
    fold_datasets = []
    # rem_dataset is remaining portion of dataset
    rem_dataset = dataset
    for fold in range(k):
      # Note starts as 1/k since fold starts at 0. Ends at 1 since fold goes up
      # to k-1.
      frac_fold = 1. / (k - fold)
      fold_dir = directories[fold]
      fold_inds, rem_inds, _ = self.split(
          rem_dataset,
          frac_train=frac_fold,
          frac_valid=1 - frac_fold,
          frac_test=0)
      fold_dataset = rem_dataset.select(fold_inds, fold_dir)
      rem_dir = tempfile.mkdtemp()
      rem_dataset = rem_dataset.select(rem_inds, rem_dir)
      fold_datasets.append(fold_dataset)
    return fold_datasets
```

**Comment by rbharath** [comment]
I like the proposed change. PRs welcome

**Comment by lilleswing** [comment]
ok I'll do it -- but this is a BREAKING change for people using this interface.
